x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -341,6 +341,32 @@ export function getIndexPatternDatasource({
     },
     getDatasourceSuggestionsFromCurrentState,
     getDatasourceSuggestionsForVisualizeField,
+
+    getErrorMessages(state) {
+      if (state) {
+        const invalidLayers = getInvalidReferences(state);
+        if (invalidLayers.length > 0) {
+          const realIndex = Object.values(state.layers)
+            .map((layer, i) => {
+              if (invalidLayers.includes(layer)) {
+                return i + 1;
+              }
+            })
+            .filter(Boolean) as number[];
+          return [
```

Comment:
Nit: Two things about this message:
* Can we not mention layer as long as there's just a single one? It's confusing for things like pie charts
* Can we mention the problematic field name?